jupytext/contentsmanager.py

```diff
@@ -481,6 +481,7 @@ def rename_file(self, old_path, new_path):
             for old_alt_path, alt_fmt in old_alt_paths:
                 new_alt_path = full_path(new_base, alt_fmt)
                 if self.exists(old_alt_path):
+                    os.makedirs(os.path.dirname(new_alt_path), exist_ok=True)
```

Comment:
I think we should create the directory using the contents manager itself, not `os.makedirs`, because in some cases the underlying contents manager is not based on a file system.

Reply:
Hello @mwouts, that's a fair point indeed!

I am not familiar with Jupyter's internals, but I can take a look.

When talking about the "underlying contents manager", are you referring to the [FileContentsManager](https://github.com/jupyter-server/jupyter_server/blob/3ba9ac973643b56011cb6fbd63dc1093865b174e/jupyter_server/services/contents/filemanager.py#L42) from [jupyter_server](https://github.com/jupyter-server/jupyter_server/tree/3ba9ac973643b56011cb6fbd63dc1093865b174e)?

If so, I can see there is a [`_save_directory`](https://github.com/jupyter-server/jupyter_server/blob/3ba9ac973643b56011cb6fbd63dc1093865b174e/jupyter_server/services/contents/filemanager.py#L438) method, but it doesn't recursive to me.

Any pointer?